File: Assignment14/Q5.c
///////////////////////////////////////////////////////////////////
//
//      Including Header Files
//
///////////////////////////////////////////////////////////////////

#include<stdio.h>
#include<stdlib.h>

///////////////////////////////////////////////////////////////////
//
//      Function name:  Frequency
//      Input:          Integer Array, Integer, Integer
//      Output:         Integer
//      Description:    Accept array and one number from user
//                      and return frequency of that number
//      Date:           03/06/2026
//      Author:         Pratik Subhash Mahale
//
///////////////////////////////////////////////////////////////////

int Frequency(
                int Arr[],     // Input array
                int iLength,   // Number of elements
                int iNo        // Number whose frequency is to be searched
             )
{
    int iCnt = 0;              // Loop counter
    int iCount = 0;            // Stores frequency of iNo

    // Traverse the array
    for(iCnt = 0; iCnt < iLength; iCnt++)
    {
        // Check whether current element matches iNo
        if(Arr[iCnt] == iNo)
        {
            iCount++;
        }
    }

    return iCount;
}

///////////////////////////////////////////////////////////////////
//
//      Entry point function
//
///////////////////////////////////////////////////////////////////

int main()
{
    int iSize = 0;             // Stores number of elements
    int iRet = 0;              // Stores return value from function
    int iCnt = 0;              // Loop counter
    int iValue = 0;            // Number to search

    int *p = NULL;             // Pointer for dynamic memory allocation

    printf("Enter number of elements\n");
    scanf("%d",&iSize);

    printf("Enter the number\n");
    scanf("%d",&iValue);

    // Allocate memory dynamically
    p = (int *)malloc(iSize * sizeof(int));

    // Check whether memory allocation is successful
    if(p == NULL)
    {
        printf("Unable to allocate memory\n");
        return -1;
    }

    printf("Enter %d elements\n",iSize);

    // Accept elements from user
    for(iCnt = 0; iCnt < iSize; iCnt++)
    {
        printf("Enter element : %d\n",iCnt + 1);
        scanf("%d",&p[iCnt]);
    }

    // Function call
    iRet = Frequency(p, iSize, iValue);

    // Display result
    printf("%d\n",iRet);

    // Deallocate memory
    free(p);

    return 0;
}

///////////////////////////////////////////////////////////////////
//
//      Input :
//      N : 6
//
//      NO : 66
//
//      Elements :
//      85 66 3 66 93 88
//
//      Output :
//      2
//
///////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////
//
//      Input :
//      N : 6
//
//      NO : 12
//
//      Elements :
//      85 11 3 15 11 111
//
//      Output :
//      0
//
///////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////
//
//      Time Complexity : O(n)
//
//      Space Complexity : O(1)
//
//      Where n is the number of elements in array
//
///////////////////////////////////////////////////////////////////

File: Assignment19/Q5.java

///////////////////////////////////////////////////////////////////
//
//      File Name : Q5.java
//      Project   : Logic Building Assignment 19
//
///////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////
//
//      Class Name : PowerLogic
//      Method Name: calculatePower
//      Input      : Two Integers
//      Output     : Void
//      Description: Calculate power of a number
//                   using loops
//      Date       : 08/06/2026
//      Author     : Pratik Subhash Mahale
//
///////////////////////////////////////////////////////////////////

class PowerLogic
{
    void calculatePower(int base, int exp)
    {
        int i = 0;          // Loop counter
        int power = 1;      // Stores result

        // Calculate power
        for(i = 1; i <= exp; i++)
        {
            power = power * base;
        }

        // Display result
        System.out.println(base + " raised to " + exp + " = " + power);
    }
}

///////////////////////////////////////////////////////////////////
//
//      Main Class
//
///////////////////////////////////////////////////////////////////

public class Q5
{
    public static void main(String args[])
    {
        // Create object of PowerLogic class
        PowerLogic obj = new PowerLogic();

        // Function call
        obj.calculatePower(2, 5);
    }
}

///////////////////////////////////////////////////////////////////
//
//      Input  : Base = 2, Exponent = 5
//
//      Output :
//      2 raised to 5 = 32
//
///////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////
//
//      Input  : Base = 3, Exponent = 4
//
//      Output :
//      3 raised to 4 = 81
//
///////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////
//
//      Time Complexity : O(exp)
//
//      Space Complexity : O(1)
//
//      Where exp = Exponent value
//
///////////////////////////////////////////////////////////////////
